jax/numpy/lax_numpy.py

```diff
@@ -242,10 +242,64 @@ def _promote_args_like(op, *args):
 def _constant_like(x, const):
   return onp.array(const, dtype=_dtype(x))
 
+
+def update_numpydoc(docstr, tabbed, fun, op):
+  original_idx = docstr.find("Original docstring below")
+
+  if tabbed:
+    lines = docstr[original_idx + 25:].split('\n')
+    for idx, line in enumerate(lines):
+      lines[idx] = line.replace('    ', '', 1)
+    docstr = docstr[:original_idx + 25] + '\n'.join(lines)
+
+  try:
+    param_list = [arg for arg in fun.__code__.co_varnames]
+    param_list = [arg for arg in param_list if arg in op.__code__.co_varnames]
+  except AttributeError:
+    begin_idx = docstr.find(fun.__name__ + '(', original_idx) + len(fun.__name__) + 1
+    end_idx = docstr.find(')', begin_idx)
+    param_list = docstr[begin_idx:end_idx].split(', ')
+    for idx, p in enumerate(param_list):
+      eq_idx = p.find('=')
+      if eq_idx == -1:
+        eq_idx = len(p)
+      param = p[:eq_idx]
+      if param[0] == '[':
+        param = param[1:]
+      if param[-1] == ']':
+        param = param[:-1]
+      if param not in op.__code__.co_varnames:
+        param_list[idx] = ''
+    param_list = [param for param in param_list if param != '']
+    parameters = ', '.join(param_list)
+    docstr = docstr[:begin_idx] + parameters + docstr[end_idx:]
+
+  begin_idx = docstr.find("Parameters")
+  begin_idx = docstr.find("--\n", begin_idx) + 2
+  end_idx = docstr.find("Returns", begin_idx)
+
+  parameters = docstr[begin_idx:end_idx]
+  param_list = parameters.replace('\n    ', '@@').split('\n')
+  for idx, p in enumerate(param_list):
+    param = p[:p.find(' : ')].split(", ")[0]
+    if param not in op.__code__.co_varnames:
+      param_list[idx] = ''
+  param_list = [param for param in param_list if param != '']
+  parameters = '\n'.join(param_list).replace('@@', '\n    ')
+  return docstr[:begin_idx + 1] + parameters + docstr[end_idx - 2:]
+
 _numpy_signature_re = re.compile(r'^([\w., ]+=)?\s*[\w\.]+\(.*\)$')
 
-def _wraps(fun):
-  """Like functools.wraps but works with numpy.ufuncs."""
+def _wraps(fun, updates=True, tabbed=True):
+  """Like functools.wraps but works with numpy.ufuncs.
+    Parameters:
+      fun: The function being wrapped
+      updates: Whether you need to update the documentation, 
```

Comment:
> The parameter names were confusing when I saw them in usage, because _wraps does something to the function, and I thought that the new parameters refer somehow to the function computation. Perhaps "update_doc" and "tabbed_doc" would be clearer?

Good point, I will change this.

> I am also wondering if we really need the "tabbed" parameter, can't this situation be detected?

Huuuuum, I think I can try to check if there's one tab as the first character of the original docstring, I will try to implement that today or tomorrow and see if it works.

Reply:
So I believe I fixed this